core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -943,6 +944,9 @@ private ExprValue exprSubDateInterval(ExprValue date, ExprValue expr) {
    * @return ExprValue.
    */
   private ExprValue exprTime(ExprValue exprValue) {
+    if (exprValue.type() == DATE) {
+      return new ExprTimeValue("00:00:00");
+    }
```

Comment:
Is this case already handled by line 953? Because I can see `ExprDateValue.timeValue()` does return `LocalTime.of(0, 0, 0);` already.

Reply:
Fixed in [68b9a26](https://github.com/opensearch-project/sql/pull/991/commits/68b9a26fd82a7504d4ec62c0922114f9e450a8bc)